frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx

```diff
@@ -46,7 +46,7 @@ const UpdateBlockPoolModal = withHandlePromise((props: UpdateBlockPoolModalProps
   );
 
   const populateBlockPoolData = React.useCallback(
-    (poolConfig: StoragePoolKind) => {
+    (poolConfig: K8sResourceKind) => {
```

Comment:
```suggestion
    (poolConfig: K8sResourceCommon) => {
```

Reply:
Why K8sResourceCommon? K8sResourceCommon dont even have a spec key, i am using spec key inside this callback

Reply:
K8sResourceKind is not really well defined type. The spec defnition is very vague. 
```
spec?: {
    selector?: Selector | MatchLabels;
    [key: string]: any;
  };
  ```
  Rather than using this. We should define `PoolResourceKind` with exact fields that are contained in Pool Resource CR. Else we are not really leveraging typescript type checking.